dubbo-config/dubbo-config-api/src/main/java/org/apache/dubbo/config/deploy/DefaultModuleDeployer.java

```diff
@@ -92,14 +92,8 @@ public DefaultModuleDeployer(ModuleModel moduleModel) {
 
     @Override
     public void initialize() throws IllegalStateException {
-        if (initialized.get()) {
-            return;
-        }
         // Ensure that the initialization is completed when concurrent calls
-        synchronized (this) {
-            if (initialized.get()) {
-                return;
-            }
+        if (initialized.compareAndSet(false, true)) {
             loadConfigs();
```

Comment:
see `DefaultApplicationDeployer.initialize()`

Reply:
Adjust the type of initialized to volatile boolean